integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonOption.scala

```diff
@@ -84,5 +84,22 @@ class CarbonOption(options: Map[String, String]) {
 
   lazy val dateformat: Option[String] = options.get("dateformat")
 
+  lazy val SPATIAL_INDEX: Option[String] = options.get("SPATIAL_INDEX")
+
+  lazy val SPATIAL_INDEX_type: Option[String] = options.get("SPATIAL_INDEX.mygeohash.type")
+
+  lazy val SPATIAL_INDEX_sourcecolumns: Option[String] = options.get(
+    "SPATIAL_INDEX.mygeohash.sourcecolumns")
+
+  lazy val SPATIAL_INDEX_originLatitude: Option[String] = options.get(
+    "SPATIAL_INDEX.mygeohash.originLatitude")
+
+  lazy val SPATIAL_INDEX_gridSize: Option[String] = options.get("SPATIAL_INDEX.mygeohash.gridSize")
```

Comment:
what if the user doesn't provide some other required options, when spatial index property is provided. Please check and handle

Reply:
Done